### PR TITLE
Create Data Model

### PR DESCRIPTION
**Pull request by Mahdi-Niazi**
In this milestone I:

- Add users migration file 
- Add posts migration file
- Add comments migration file
- Add likes migration file

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @Mahdi-Niazi 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't follow proper gitflow patterrn , Kindly create a pull request to compare your feature branch with the development branch and submit the link to that pull request via your Dashboard.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20231130105924_create_comments.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CreateComments < ActiveRecord::Migration[7.1]
+  def change
+    create_table :comments do |t|
+      t.bigint :author_id
```
Comment (MahmoudBakr23) [inline db/migrate/20231130105924_create_comments.rb]: - [ ] According to the [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png), please keep in mind that the author_id column is only used with the posts table, and not in the comments and likes tables, please stick to the given ERD for better a development experience. 🙏

File: db/migrate/20231130110318_create_likes.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class CreateLikes < ActiveRecord::Migration[7.1]
+  def change
+    create_table :likes do |t|
+      t.bigint :author_id
```
Comment (MahmoudBakr23) [inline db/migrate/20231130110318_create_likes.rb]: - [ ] According to the [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png), please keep in mind that the author_id column is only used with the posts table, and not in the comments and likes tables, please stick to the given ERD for better a development experience. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @Ochiengsteven 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on making the migrations' files correctly so far
- Nice work with the indexes
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @Mahdi-Niazi, 
congrats 2

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Refactored Tables As Requested :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**